### ISSUE TITLE
Representation model output varies with length in each batch

### ISSUE THREAD

**Issue by ryan-224** [ThilinaRajapakse/simpletransformers#815]
Hi,

Thanks for sharing this great work with us.

I was trying to extract the embeddings/representation with the RepresentationModel and use the output for my down-stream task.

I noticed for the same sentence, when it appeared in different batches with differences in longest sentence length, the encoder's output were different.

Here is a simple illustration:
For the same sentence "I like you" appeared in different batches, the embeddings are different.
![image](https://user-images.githubusercontent.com/58620549/98898609-f9fc3480-24e8-11eb-8255-2d6d5dd6dcea.png)

May I know if there is any suggestion dealing with it?

Thanks for your help!

**Comment by ThilinaRajapakse** [comment]
Does this happen with a fine-tuned model? If it's with a pre-trained model without fine-tuning, I suspect that the extra padding tokens cause the representation to change.

**Comment by ryan-224** [comment]
Hi,

This happens with a pre-trained model.

**Comment by ThilinaRajapakse** [comment]
The `RepresentationModel` wasn't using the attention mask previously but I fixed that. However, this issue still seems to be happening. I can't figure out why.

**Comment by nikirg** [comment]
I have the same problem

```python
embedder = RepresentationModel(
    model_type="bert", 
    model_name="DeepPavlov/rubert-base-cased"
)

embeddings = embedder.encode_sentences(
    dataset.input_text, 
    combine_strategy="mean"
)

print(embeddings [0])
```
array([ 1.67089075e-01,  9.64678153e-02, -3.61526534e-02,  1.71094999e-01,
        7.29178190e-02, -1.22159965e-01,  1.88725609e-02,  5.42912818e-02, ... etc.

and 
```python
embedder.encode_sentences(
    dataset.input_text.iloc[:2], 
    combine_strategy="mean"
)[0]
```
output:
array([ 1.23940028e-01,  9.66407545e-03, -7.81051815e-02,  2.30304316e-01,
        1.17382117e-01, -1.54828295e-01,  1.16960453e-02,  1.30336598e-01, ... etc.
